posthog/hogql_queries/insights/paths_query_runner.py

```diff
@@ -126,16 +129,113 @@ def construct_event_hogql(self) -> ast.Expr:
 
         return event_hogql
 
+    def handle_funnel(self) -> tuple[list, Optional[ast.Expr]]:
+        if (
+            not self.query.pathsFilter.funnelActorsQuery
+            or not self.query.pathsFilter.funnelActorsQuery.source.funnelsFilter
+        ):
+            raise ValueError("Funnel actors query is required for funnel paths")
+
+        if self.query.pathsFilter.funnelPaths in (
+            FunnelPathType.funnel_path_after_step,
+            FunnelPathType.funnel_path_before_step,
+        ):
+            funnel_fields = [
+                ast.Alias(alias="target_timestamp", expr=ast.Field(chain=["funnel_actors", "timestamp"])),
+            ]
+            interval = self.query.pathsFilter.funnelActorsQuery.source.funnelsFilter.funnelWindowInterval or 14
+            unit = self.query.pathsFilter.funnelActorsQuery.source.funnelsFilter.funnelWindowIntervalUnit
+            interval_unit = funnel_window_interval_unit_to_sql(unit)
+            operator = ">=" if self.query.pathsFilter.funnelPaths == FunnelPathType.funnel_path_after_step else "<="
+            default_case = f"events.timestamp {operator} {{target_timestamp}}"
+            if (
+                self.query.pathsFilter.funnelPaths == FunnelPathType.funnel_path_after_step
+                and self.query.pathsFilter.funnelActorsQuery.funnelStep
+                and self.query.pathsFilter.funnelActorsQuery.funnelStep < 0
+            ):
+                default_case += f" + INTERVAL {interval} {interval_unit}"
```

Comment:
Just the other day I noticed an issue with HogQL and adding intervals. Basically ClickHouse requires the date to be in UTC, otherwise it will have an extra/missing hour for periods spanning a DST transition. See my fix and associated tests here: https://github.com/PostHog/posthog/pull/21042. The solution for this is to wrap the relevant date in `toTimeZone(mydate, 'UTC')`